### ISSUE TITLE
Interface smartcast doesn't work with selector expr

### ISSUE THREAD

**Issue by danieldaeschle** [vlang/v#7039]
**What did you do?**
```v
module main

interface Foo {
	test()
}

struct Bar {
	x int
}

fn (b Bar) test() {}

struct Wrapper {
	foo Foo
}

fn main() {
	w := Wrapper{}
	if w.foo is Bar {
		_ := w.foo.x
	}
}
```

**What did you expect to see?**
It should smartcast it
 
**What did you see instead?**
```
main.v:20:14: error: `Foo` is not a struct
   18 |     w := Wrapper{}
   19 |     if w.foo is Bar {
   20 |         _ := w.foo.x
      |                    ^
   21 |     }
   22 | }
```

**Comment by danieldaeschle** [comment]
And we should get rid of the `smartcast` property used in code for idents.

**Comment by spaceface777** [comment]
This was fixed a while ago (see #9256).
